testcase/testcase.go

```diff
@@ -309,34 +315,24 @@ func marshalToFile(event logstash.Event, filename string) error {
 // path and optional arguments) and returns whether the files were
 // equal, i.e. whether the diff command returned a zero exit
 // status. The returned error value will be set if there was a problem
-// running the command. If quiet is true, the output of the diff
-// command will be discarded. Otherwise the child process will inherit
-// stdout and stderr from the parent.
-func runDiffCommand(command []string, file1, file2 string, quiet bool) (bool, error) {
+// running the command. The output of the diff command will is returned
+// as string.
+func runDiffCommand(command []string, file1, file2 string) (bool, string, error) {
 	fullCommand := append(command, file1)
 	fullCommand = append(fullCommand, file2)
 	c := exec.Command(fullCommand[0], fullCommand[1:]...)
-	if !quiet {
-		c.Stdout = os.Stdout
-		c.Stderr = os.Stderr
-	}
+	var b bytes.Buffer
+	c.Stdout = &b
+	c.Stderr = &b
```

Comment:
This doesn't actually work. `b` is empty and the child just inherits the parent's stdout and stderr. Use c.CombinedOutput() instead, or c.Run() instead of c.Start() followed by c.Wait()?